pages/api/addContact.ts

```diff
@@ -66,16 +67,21 @@ export default async function handler (
       }
     }
 
-    // No return value; if it fails, it'd have thrown exception
-    await managers.contactDBManager.addContact(req.body.userid, contactToAdd)
-
-    // Gets the contact we just added in and sends it back
-    res.status(404).json({
-      message: await managers
-        .contactDBManager
-        .getContacts(req.body.userid)
-        .then((arr) => arr[arr.length - 1])
-    })
+    const user = await userManager.getUserByEmail(req.body.email)
```

Comment:
I don't think you authenticate the user, you need to call the verification method to ensure that the token in the authorization header matches the session field in the user database for this user.

Reply:
check line 21